physical/raft/transaction.go

```diff
@@ -715,6 +720,7 @@ func (t *RaftTransaction) Commit(ctx context.Context) error {
 	// the transaction application in Raft, applyLog will gather it for us
 	// and return it as a proper error.
 	t.b.l.RLock()
+	log.LowestActiveIndex = pointerutil.Ptr(min(lowestActiveIndex, t.b.raft.AppliedIndex()-1)) // we need to cap the lowest active index, otherwise we might miss transaction started later
```

Comment:
No sure, if we really need this. Seems to be a pathological case, especially on clusters with some load (as this will only affect transactions that are committed within their start index and every single log bumps the index).

I guess it is the more "correct" way to handle it, but it has the downside, that it might keep entries longer in the tracker than needed. If we on the other hand evict entries to early, we will have additional reads. As this should only happen on clusters without much load, it probably does not matter to have a few additional reads.